src/Discovery.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bundle\MercureBundle;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\WebLink\GenericLinkProvider;
+use Symfony\Component\WebLink\Link;
+
+final class Discovery
```

Comment:
```suggestion
final class DiscoveryUtils
```

Or we should rename `AuthUtils` in just `Auth`?

Also, this class could be moved to the component.